core/src/main/java/org/apache/struts2/interceptor/I18nInterceptor.java

```diff
@@ -281,15 +281,17 @@ public Locale find() {
 
         @Override
         public Locale store(ActionInvocation invocation, Locale locale) {
-            //save it in session
-            Map<String, Object> session = invocation.getInvocationContext().getSession();
+            HttpSession session = ServletActionContext.getRequest().getSession(false);
 
             if (session != null) {
-                String sessionId = ServletActionContext.getRequest().getSession().getId();
+                String sessionId = session.getId();
                 synchronized (sessionId.intern()) {
-                    session.put(attributeName, locale);
+                    invocation.getInvocationContext().getSession().put(attributeName, locale);
                 }
+            } else {
+                LOG.debug("session creation avoided as it doesn't exist already");
             }
+
             return locale;
```

Comment:
When `shouldStore` field is not false (i.e. we should store), shouldn't we forcefully create a session and store the locale? if we should not, shouldn't we return `null` here?

Reply:
Maybe we should forcefully create the session but I would leave that to a user, if there was no session just ignore (a user fault). And returning `null` can break action processing as then the null will be set in `ActionContext`

Reply:
👍